### PR TITLE
feat(utils): add amplitude for module interaction page

### PR DESCRIPTION
**Pull request by sunzsolo**

### CODE REVIEW

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/interact/component/common/FunctionAccordion.tsx [hunk 1]
```diff
@@ -26,36 +28,55 @@ export const FunctionAccordion = ({
   triggerText,
   selectedFn,
   setSelectedFn,
+  amptracktab,
```
Comment (evilpeach) [inline src/lib/pages/interact/component/common/FunctionAccordion.tsx:31]: amptractTab

File: src/lib/pages/interact/component/common/FunctionAccordion.tsx [hunk 2]
```diff
@@ -26,36 +28,55 @@ export const FunctionAccordion = ({
...
-    </AccordionPanel>
+    {({ isExpanded }) => (
+      <>
+        <h6>
+          <AccordionButton
+            onClick={() => {
+              trackUseExpand({
+                action: isExpanded ? "collapse" : "expand",
+                component: "module_interaction_function_accordion",
+                info: { Tab: amptracktab },
```
Comment (evilpeach) [inline src/lib/pages/interact/component/common/FunctionAccordion.tsx:42]:
```suggestion
                info: { tab: amptrackTab },
```

File: src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx [hunk 3]
```diff
@@ -86,9 +87,10 @@ export const ModuleSelectorInput = ({
     (event: KeyboardEvent<HTMLInputElement>) => {
       if (event.key === "Enter") {
         handleSubmit();
+        trackUseModuleSelectionInputFill(addr, !!moduleName, !!functionName);
```
Comment (evilpeach) [inline src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx:90]: move to `handleSubmit` fn

File: src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx [hunk 4]
```diff
@@ -86,9 +87,10 @@ export const ModuleSelectorInput = ({
...
       }
     },
-    [handleSubmit]
+    [functionName, handleSubmit, moduleName, addr]
```
Comment (evilpeach) [inline src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx:93]: remove unused

File: src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx [hunk 5]
```diff
@@ -116,7 +118,14 @@ export const ModuleSelectorInput = ({
       <Flex gap={2}>
         <Button
           variant="primary"
-          onClick={handleSubmit}
+          onClick={() => {
+            handleSubmit();
+            trackUseModuleSelectionInputFill(
```
Comment (evilpeach) [inline src/lib/pages/interact/component/drawer/selector/ModuleSelectorInput.tsx:123]: ditto

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

File: src/lib/pages/interact/component/common/FunctionAccordion.tsx [hunk 6]
```diff
@@ -17,6 +18,7 @@ interface FunctionAccordionProps {
   isEmpty: boolean;
   triggerText: string;
   selectedFn: Option<ExposedFunction>;
+  amptrackTab: string;
```
Comment (songwongtp) [inline src/lib/pages/interact/component/common/FunctionAccordion.tsx:21]: move below `setSelectedFn`

File: src/lib/pages/interact/component/common/FunctionAccordion.tsx [hunk 7]
```diff
@@ -26,36 +28,55 @@ export const FunctionAccordion = ({
...
+                    setSelectedFn(fn);
+                  }}
+                />
+              ))
+            ) : (
+              <NoImageEmptyState
+                desc={isEmpty ? "No function" : "No matching function found"}
+              />
+            )}
+          </Flex>
```
Comment (songwongtp) [inline src/lib/pages/interact/component/common/FunctionAccordion.tsx:77]: move `</Flex>` to between line 71 and 72

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Pull request merged**